java/ql/test/query-tests/security/CWE-312/AndroidManifest.xml

```diff
@@ -4,7 +4,7 @@
     android:versionCode="1"
     android:versionName="0.1" >
 
-    <application>
+    <application android:allowBackup="true">
```

Comment:
I would expect this to need a `<!-- $ hasAllowedBackupEnabled -->` comment to pass the test.

Also, instead of mixing this test with the existing one about cleartext storage, I would create a new folder under `CWE-312` (something like `backupAllowed`) and place all the files related to this query in there. That way you can also add other `AndroidManifest.xml` files to test the implicitly allowed and explicitly disallowed cases.

You can take inspiration from https://github.com/github/codeql/pull/9939.